Kitodo-FileManagement/src/main/java/org/kitodo/filemanagement/locking/LockRequests.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.filemanagement.locking;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.kitodo.api.filemanagement.LockingMode;
+
+/**
+ * Capsules a user request for locks.
+ */
+class LockRequests {
+    /**
+     * Which locks the user would like.
+     */
+    private final Map<URI, LockingMode> locks;
+
+    /**
+     * Which user requests the locks.
+     */
+    private final String user;
+
+    /**
+     * Create a new request object.
+     *
+     * @param user
+     *            user making the request
+     * @param locks
+     *            which locks he or she would like
+     */
+    LockRequests(String user, Map<URI, LockingMode> locks) {
+        this.user = user;
+        this.locks = locks;
+    }
+
+    /**
+     * Compares two lock objects.
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        LockRequests other = (LockRequests) obj;
+        if (locks == null) {
+            if (other.locks != null) {
+                return false;
+            }
+        } else if (!locks.equals(other.locks)) {
+            return false;
+        }
+        if (user == null) {
+            if (other.user != null) {
```

Comment:
IF can be simplified to one statement.

Reply:
No, this only works at first glance, but otherwise the else if block would also be executed if both objects are null (and would result in a NullPointerException). By the way: This code is generated automatically by Eclipse. I assume the underlying method was reviewed by a bunch of computer science professors worldwide. In short, that works, we do not need to review that.